lib/src/components/NxNavigationSidebar/NxNavigationSidebar.scss

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+@import '../../scss-shared/nx-variables';
+@import '../../scss-shared/nx-text-helpers';
+@import '../../scss-shared/nx-container-helpers';
+
+$nx-sidebar-max-width: 300px;
+$nx-sidebar-width-open: 250px;
+$nx-sidebar-width-closed: 72px;
+$nx-sidebar-logo-height: 34px;
+
+$nx-sidebar-text-color: $nx-white;
+$nx-sidebar-background-color: $nx-blue-1000;
+
+$nx-sidebar-active-color: $nx-indigo-400;
+$nx-sidebar-selected-color: $nx-indigo-800;
+$nx-sidebar-hover-color: $nx-indigo-600;
+
+$nx-sidebar-focus-border: 1px solid $nx-focus-border-color;
+
+$nx-sidebar-separator-color: $nx-grey-900;
+
+.nx-page-sidebar-operable {
+  @include container-vertical;
+
+  background-color: $nx-sidebar-background-color;
+  box-sizing: border-box;
+  color: $nx-sidebar-text-color;
+  max-width: $nx-sidebar-max-width;
+  padding: $nx-spacing-xl $nx-spacing-l $nx-spacing-l $nx-spacing-l;
+  width: $nx-sidebar-width-open;
+
+  .nx-text-link {
+    color: $nx-sidebar-text-color;
+  }
+}
+
+.nx-page-sidebar-operable__header {
+  display: flex;
+  margin-bottom: $nx-spacing-xl;
+}
+
+.nx-page-sidebar-operable__toggle {
+  border-color: transparent;
+  border-radius: 50%;
+  color: $nx-sidebar-text-color;
+  flex: none;
+  height: 34px;
+  margin: 0;
+  width: 34px;
+
+  &:focus {
+    border-color: $nx-focus-border-color;
+  }
+
+  &:active {
+    background-color: $nx-sidebar-active-color;
+  }
+
+  &:hover {
+    background-color: $nx-sidebar-hover-color;
+  }
+}
+
+.nx-page-sidebar-operable__product-info {
+  flex: 1 1 auto;
+  height: $nx-sidebar-logo-height;
+}
+
+.nx-sidebar-navigation {
+  border-bottom: 1px solid $nx-sidebar-separator-color;
+  flex: none;
+
+  margin: 0 -#{$nx-spacing-l} $nx-spacing-md -#{$nx-spacing-l};
+  padding-bottom: $nx-spacing-xxl;
+}
+
+.nx-sidebar-navigation__link {
+  @include container-horizontal;
+
+  align-items: center;
+  border: 1px solid transparent;
+  display: flex;
+  padding: 0 $nx-spacing-l;
+
+  .nx-icon {
+    margin: $nx-spacing-xs $nx-spacing-xs $nx-spacing-xs 0;
+  }
+
+  &:focus {
+    border-color: $nx-focus-border-color;
+    box-shadow: $nx-focus-box-shadow;
+    outline: none;
+  }
+
+  &:hover {
+    background-color: $nx-sidebar-hover-color;
+  }
+
+  &:active {
+    background-color: $nx-sidebar-active-color;
+  }
+
+  &.selected {
+    @include semi-bold;
+    background-color: $nx-sidebar-selected-color;
+  }
+}
+
+.nx-sidebar-navigation__text {
+  @include nx-truncate-ellipsis;
+}
+
+.nx-page-sidebar-operable.open {
+  .nx-page-sidebar-operable__collapsed-content {
+    display: none;
+  }
+}
+
+.nx-page-sidebar-operable.closed {
+  margin: 0;
+  padding-left: 0;
+  padding-right: 0;
```

Comment:
Similarly to why I thought it should have left/right padding when open, shouldn't it also when it is closed, for the same reasons?

Reply:
> Similarly to why I thought it should have left/right padding when open, shouldn't it also when it is closed, for the same reasons?

Fixed in 6d24ef22a90b4066e77bebfb518f7506b1183566

Reply:
> Fixed in 6d24ef2

Looking at the mockups it isn't clear to me if the collapsed sidebar's padding is supposed to be the same thickness as in the expanded sidebar.  In particular around the word "release" in the custom content at the bottom, there is only 14px on either side.  @dturner32 ?

Reply:
In particular around the word "release" in the custom content at the bottom

I wouldn't mind that too much — it is content that should be implemented elsewhere and is subject to the consumer's design approval. We could keep our current paddings and then let the consumer use whatever they need to position that according to their mocks.

Reply:
> We could keep our current paddings and then let the consumer use whatever they need to position that according to their mocks.

We do want a reasonable default though. With 24px of padding on each side that only leaves 24px of content width.  Most anything aside from an icon button is going to exceed that.

Reply:
> > We could keep our current paddings and then let the consumer use whatever they need to position that according to their mocks.
> 
> We do want a reasonable default though. With 24px of padding on each side that only leaves 24px of content width. Most anything aside from an icon button is going to exceed that.

Isn't the collapsed sidebar intended to be mostly buttons-only? Based on what I've seen in the mocks —current and past— I think those release texts are the exception and not the rule.

Reply:
> > Fixed in [6d24ef2](https://github.com/sonatype/sonatype-react-shared-components/commit/6d24ef22a90b4066e77bebfb518f7506b1183566)
> 
> Looking at the mockups it isn't clear to me if the collapsed sidebar's padding is supposed to be the same thickness as in the expanded sidebar. In particular around the word "release" in the custom content at the bottom, there is only 14px on either side. @dturner32 ?

I think the direction forward for the collapsed sidebar should be 'icons only'. The padding should remain at 24px on left and right whether its open or closed (which I can call out in the mock). This means that the release text wouldn't fit in that space. I can remove that and just make a note that when the sidebar is collapsed, only the button icons will appear.

Reply:
just updated the zeplin mock

Reply:
> I think the direction forward for the collapsed sidebar should be 'icons only'. The padding should remain at 24px on left and right whether its open or closed (which I can call out in the mock). This means that the release text wouldn't fit in that space. I can remove that and just make a note that when the sidebar is collapsed, only the button icons will appear.

I'd say there are still two potential issues here:
1. the collapse/expand button is 32px wide but the content area of the collapsed sidebar is only 22px wide (not 24 as I mistakenly said previously).  This means the button borders overflow into the padding area.  We could say that we are simply fine with that.  It is centered at least, due to `justify-content` on the header.
2. The nav icons are not quite centered.  The icons are only 20px wide while the space allotted to them is, again, 22px wide.  So using some form of center alignment is necessary here as well.

Reply:
> This means the button borders overflow into the padding area. 

I have no issue with that. Like you said, at least it is centered.

> 2\. The nav icons are not quite centered.  The icons are only 20px wide while the space allotted to them is, again, 22px wide.  So using some form of center alignment is necessary here as well.

Like this? cb4f16634620392251fc10fa7d69b053426d3ac4